### PR TITLE
[client] Cleanup firewall state on startup

### PR DESCRIPTION
**Pull request by lixmal**
### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: client/firewall/create.go [hunk 1]
```diff
@@ -11,10 +11,11 @@ import (
 
 	firewall "github.com/netbirdio/netbird/client/firewall/manager"
 	"github.com/netbirdio/netbird/client/firewall/uspfilter"
+	"github.com/netbirdio/netbird/client/internal/statemanager"
 )
 
 // NewFirewall creates a firewall manager instance
-func NewFirewall(context context.Context, iface IFaceMapper) (firewall.Manager, error) {
+func NewFirewall(_ context.Context, iface IFaceMapper, _ *statemanager.Manager) (firewall.Manager, error) {
```
Comment (pappz) [inline client/firewall/create.go]: You can fully remove context.Context parameter from here and from the create_linux.go too. It is unused.

File: client/firewall/create_linux.go [hunk 2]
```diff
@@ -58,6 +59,12 @@ func NewFirewall(context context.Context, iface IFaceMapper) (firewall.Manager,
 		log.Info("no firewall manager found, trying to use userspace packet filtering firewall")
 	}
 
+	if fm != nil {
+		if err := fm.Init(stateManager); err != nil {
```
Comment (pappz) [inline client/firewall/create_linux.go]: What will be if the **iface.IsUserspaceBind()** scope happens an error? Should not we revert this initialization?

Comment (lixmal) [reply client/firewall/create_linux.go]: Probably, but this is out of scope for the PR. I just split out the init logic in the `Create` functions so it is not run when cleaning up only.

File: client/firewall/create.go (hunk 1 above)
Comment (lixmal) [reply client/firewall/create.go]: done

**Review by pappz — APPROVED** [review]

**Pull request merged**